test/framework/src/main/java/org/elasticsearch/test/ESAllocationTestCase.java

```diff
@@ -157,15 +158,17 @@ protected  static ClusterState startRandomInitializingShard(ClusterState cluster
     }
 
     protected static AllocationDeciders yesAllocationDeciders() {
-        return new AllocationDeciders(Settings.EMPTY, new AllocationDecider[] {new TestAllocateDecision(Decision.YES)});
+        return new AllocationDeciders(Settings.EMPTY, new AllocationDecider[] {new TestAllocateDecision(Decision.YES),
```

Comment:
yeah, without it the assertions in the routing nodes universe complain that a shard is assigned to a node that already has it. I do think that we should hard code this allocation decider (and the one with replica follow primary) into the set of deciders so it will always be there. It's an integral part of how our system works